server/aesdsocket.c

```diff
@@ -0,0 +1,269 @@
+	if (server_socket_fd == -1)
+	{
+		syslog(LOG_ERR, "Error: socket() with code, %d", errno);
+	}
+	
+	status = setsockopt(server_socket_fd, SOL_SOCKET, SO_REUSEADDR, &yes, sizeof(yes));
+	if (status != 0)
+	{
+		syslog(LOG_ERR, "Error: setsocketopt() with code, %d", errno);
+	}
+	
+	
+	status = bind(server_socket_fd, server_info->ai_addr, server_info->ai_addrlen);
+	if (status != 0)
+	{
+		syslog(LOG_ERR, "Error: bind() with code, %d", errno);
+	}
+	
+	
+	if (daemon_flag == 1)
+	{
+		if (daemon_creation() != 0)
+		{
+			syslog(LOG_ERR, "Daemon creation failed");
+		}
+	}
+		
+	status = listen(server_socket_fd, 10);
+	if (status != 0)
+	{
+		syslog(LOG_ERR, "Error: listen() with code, %d", errno);
+	}
+	
+	freeaddrinfo(server_info); 
+	
+	their_size = sizeof(their_addr);
+
+	while(1)
+	{
+		client_socket_fd = accept(server_socket_fd, (struct sockaddr *)&their_addr, &their_size);
+		if (client_socket_fd == -1)
+		{
+			syslog(LOG_ERR, "Error: accept() with code, %d", errno);
+		}
+		
+      		char address_ip[20]; 
+       	const char* ip;
+       	ip = inet_ntop(AF_INET, &their_addr.sin_addr, address_ip, sizeof(address_ip));
+       	syslog(LOG_DEBUG, "Accepted connection from : %s \n" , ip);
+
+  		memset(buffer, '\0', SIZE);
+  		int num = 0;
+  		int total_buffer_length = 0;	
+  		int old_length = SIZE;  
+  		received = 0;
+  		
+  		while(!received)
+  		{
+  			received_length = 0;
+  			recv(client_socket_fd, &buffer, SIZE, 0);
+  			//ERROR
+  		 			
+  			for (int i=0; i<SIZE; i++)
+  			{
+  				received_length++;
+  				if (buffer[i] == '\n')
+  				{
+  					received = 1;
+  					break;
+  				}
+  			}
+  			
+  			if (num == 0)
+  			{
+  				total_buffer = (char *)malloc(received_length);
+  				if (total_buffer == NULL)
+  				{
+  					syslog(LOG_ERR, "malloc");
+  				}
+  				memset(total_buffer, '\0', received_length);
+  				total_buffer_length = received_length;  				
+  			}
+  			else
+  			{
+  				char *ptr = realloc(total_buffer, old_length+received_length);
+  				if (ptr == NULL)
+  				{
+  					syslog(LOG_ERR, "realloc");
+  				}
+  				else 
+  				{
+  					total_buffer = ptr;
+  					old_length += received_length;
+  				}
+  				total_buffer_length = old_length;
+  			}	
+  			memcpy(total_buffer + (num * SIZE), buffer, received_length);
+  			num++;
+  		}
+  		
+  		lseek(socket_file_fd, 0, SEEK_END);
+  			
+  		int write_return = write(socket_file_fd, total_buffer, total_buffer_length);
+  		if (write_return == -1)
+  		{
+  			syslog(LOG_ERR, "Error: write() with code, %d", errno);
+  		
+  		}
+  		
+  		else if (write_return != total_buffer_length)
+  		{
+  			syslog(LOG_ERR, "Error: partial write with code, %d", errno);
+  		}
+  		
+  		data_count += write_return;
+  			
+  		memset(buffer, '\0', SIZE);
+  		lseek(socket_file_fd, 0, SEEK_SET);
+  		  			
+  		char *buffer_read = (char *)malloc(data_count);
```

Comment:
@anuh7 what assumption are you making here about the size of the process heap versus the total size of all packets sent?